core/src/main/java/com/scalar/db/transaction/consensuscommit/ConsensusCommitMutationOperationChecker.java

```diff
@@ -0,0 +1,96 @@
+package com.scalar.db.transaction.consensuscommit;
+
+import com.scalar.db.api.ConditionalExpression;
+import com.scalar.db.api.Delete;
+import com.scalar.db.api.DeleteIf;
+import com.scalar.db.api.DeleteIfExists;
+import com.scalar.db.api.Mutation;
+import com.scalar.db.api.MutationCondition;
+import com.scalar.db.api.Operation;
+import com.scalar.db.api.Put;
+import com.scalar.db.api.PutIf;
+import com.scalar.db.api.PutIfExists;
+import com.scalar.db.api.PutIfNotExists;
+import com.scalar.db.exception.storage.ExecutionException;
+import java.util.Collections;
+import java.util.List;
+import javax.annotation.concurrent.ThreadSafe;
+
+@ThreadSafe
+public class ConsensusCommitMutationOperationChecker {
+
+  private final TransactionTableMetadataManager transactionTableMetadataManager;
+
+  public ConsensusCommitMutationOperationChecker(
+      TransactionTableMetadataManager transactionTableMetadataManager) {
+    this.transactionTableMetadataManager = transactionTableMetadataManager;
+  }
+
+  private TransactionTableMetadata getTableMetadata(Operation operation) throws ExecutionException {
+    TransactionTableMetadata metadata =
+        transactionTableMetadataManager.getTransactionTableMetadata(operation);
+    if (metadata == null) {
+      throw new IllegalArgumentException(
+          "The specified table is not found: " + operation.forFullTableName().get());
+    }
+    return metadata;
+  }
+
+  public void check(Mutation mutation) throws ExecutionException {
```

Comment:
I think we can throw `IllegalArgumentException` instead of `ExecutionException` for the operation check.

Reply:
Just to be sure we are on the same page. It is not documented but `check` only throws an `ExceptionException` if the table metadata fails to be retrieved. https://github.com/scalar-labs/scalardb/blob/0212c8b2d8bf773600383687e59f386ce5260faf/core/src/main/java/com/scalar/db/transaction/consensuscommit/ConsensusCommitMutationOperationChecker.java#L30-L31
For validation errors, `IllegalArgumentException` is thrown. 
Do you think we should catch the `ExecutionException` and rethrow it as an `IllegalArgumentException`?

Reply:
Ah sorry, I misunderstood. I thought `ExecutionException` was also thrown for validation errors. Please ignore the comment.

Reply:
I added Javadoc since it's difficult to understand why the `ExecutionException` is thrown.
https://github.com/scalar-labs/scalardb/pull/873/commits/4e05db8de6de4d94bc81de22ad3c3b40bfd7a321